### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by javdrher**
Over the past weeks I have done some basic work to set up documentation (#6). I based myself on the structure of the GPflow docs. I'm sure its still full of typos and poor grammar so this work isn't done, however feedback could be extremely important. I went over the code to introduce docstrings, made some notebooks and set up sphinx (dependencies for building the docs should match with GPflow). 

To build the docs:
`python setup.py build_sphinx` which produces a build/sphinx/html folder.

### CODE REVIEW

File: GPflowOpt/acquisition.py [hunk 1]
```diff
@@ -58,11 +73,18 @@ def _optimize_models(self):
...
-                    print("Warning - optimization restart {0}/{1} failed".format(i + 1, self._optimize_restarts))
+                    print("Warning: optimization restart {0}/{1} failed".format(i + 1, self._optimize_restarts))
             best_idx = np.argmin(map(lambda r: r.fun, runs))
             model.set_state(runs[best_idx].x)
 
     def _build_acquisition_wrapper(self, Xcand, gradients=True):
+        """
+        Build the graph to compute the acquisition function.
+
+        :param Xcand: candidate points to compute the acquisition function for
```
Comment (nknudde) [inline GPflowOpt/acquisition.py]: candidate points for which to compute the acquisition function

File: doc/source/intro.rst [hunk 2]
```diff
@@ -2,11 +2,32 @@
 Introduction
 ------------
 
+GPflowOpt is a library for Bayesian Optimization with GPflow. It makes use of TensorFlow for computation of acquisition functions,
+to offer scalability, and avoid implementation of gradients. The package was created, and is currently maintained
+by `Joachim van der Herten <http://sumo.intec.ugent.be/jvanderherten>`_ and `Ivo Couckuyt <http://sumo.intec.ugent.be/icouckuy>`_
+
+Currently the software is pre-release and under construction, hence it lacks a lot of functionality and testing. This documentation
+might also be incomplete and under development. The project is open source: if you feel you have some relevant skills and are interested in contributing then please contact us on github.
```
Comment (icouckuy) [inline doc/source/intro.rst]: might -> is also

Comment (icouckuy) [inline doc/source/intro.rst]: contact us on github by opening an issue or pull request.

-> add a direct link to the GPFlowOpt project.

File: doc/source/intro.rst [hunk 3]
```diff
@@ -2,11 +2,32 @@
...
+
 Install
 --------
-1. Install TensorFlow.
-Please see instructions on the main TensorFlow `webpage <https://www.tensorflow.org/versions/r1.0/get_started/get_started>`_. You will need version 1.0. . We find that for many users pip installation is the fastest way to get going.
+1. Install TensorFlow
+
+You will need version 1.0 or above. We find that for many users pip installation is the fastest way to get going.
+``pip install tensorflow``
```
Comment (icouckuy) [inline doc/source/intro.rst]: or for the gpu version,
``pip install tensorflow-gpu``

?

File: doc/source/intro.rst [hunk 4]
```diff
@@ -2,11 +2,32 @@
...
+A different way to install the package: ``python setup.py install`` (untested)
+
+You can run the tests with ``python setup.py test``.
+
+To build the documentation, first install extra dependencies with ``pip install .[docs]``, then proceed with ``python setup.py build_sphinx``.
+
+Acknowledgements
+-----------------
+Joachim van der Herten and Ivo Couckuyt are Ghent University - imec postdoctoral fellows. Ivo Couckuyt is supported
+by FWO vlaanderen.
```
Comment (icouckuy) [inline doc/source/intro.rst]: Vlaanderen.

**Comment by javdrher** [comment]
This seems good as a starting point for documentation. Further improvements are required as the project evolves.

**Pull request merged**